src/components/Select/common/types.ts

```diff
@@ -74,7 +74,11 @@ interface InternalSelectProps
   onOpenChange: (open: boolean) => void;
   value: Array<string>;
   sortable?: boolean;
-  onSelect: (value: string, type?: SelectionType) => void;
+  onSelect: (
+    value: string,
+    type?: SelectionType,
+    evt?: KeyboardEvent<HTMLElement>
```

Comment:
I'd suggest passing event regardless of the event type, to keep api consistent. Looks weird to me that we handle one specific case. We still should be able to tell if the change was triggered by keyboard with smth like `event instanceof KeyboardEvent`

Reply:
What do you mean by “the only typeof event we support”?
`onSelect` is also called here for example, and in this case it's `MouseEvent`
https://github.com/ClickHouse/click-ui/blob/main/src/components/Select/common/InternalSelect.tsx#L554

I'll unresolve this thread for the time being, want to understand

Reply:
>What do you mean by “the only typeof event we support”?

The only time this event is passed in to `onSelect` is when it's a `KeyboardEvent` in the `onKeyDown` handler in `InternalSelect`. So by typing it as something other than a `KeyboardEvent`, it's not really describing the behavior that exists currently accurately.

Reply:
Ok, I understand that. It might've been an unfortunate line to leave a comment to, because my point is not about types, but rather about the behavior. Should we pass the event in other instances as well? To keep component API consistent?

Reply:
Oh, I totally misunderstood. I think that makes sense.